src/contract/table.ts

```diff
@@ -104,27 +101,25 @@ export class Table<TableRow extends ABISerializableConstructor = ABISerializable
      *  - `limit`: Maximum number of rows to return.
      * @returns {TableCursor<TableRow>} Promise resolving to a `TableCursor` of the filtered table rows.
      */
-    query(
-        query: Query,
-        {limit, scope = this.contract.account, index, key_type}: QueryOptions = {}
-    ): TableCursor<TableRow> {
-        const {from, to} = query
+    query(query: Query): TableCursor<TableRow> {
+        const {from, to, limit} = query
 
         const tableRowsParams = {
             table: this.name,
             code: this.contract.account,
-            scope,
+            scope: query.scope || this.contract.account,
             type: this.rowType,
             limit: limit || this.defaultRowLimit,
             lower_bound: wrapIndexValue(from),
             upper_bound: wrapIndexValue(to),
-            key_type: key_type,
+            key_type: query.key_type,
         }
 
         return new TableCursor({
+            maxRows: limit || this.defaultRowLimit,
```

Comment:
Using the word `maxRows` is not a bad idea since it actually functions a bit differently from the get_table_rows `limit` field, but I think that it should be used in the `query()` method as well (not just the `all()` method).

Reply:
Another wild idea is maybe we get rid of `limit` all together, since you can just pass a per-request limit value into the `.next()` call.

Reply:
Ok so I think that we just had a different idea of how the query `limit` parameter should work. I imagined that it would work similarly to how the maxRows field works (which is why I was confused when you decided to use one variable name for the `first()` method and another one for the `next()`method). I see why you wanted to treat `limit` as the maximum number of rows per API call though since that's how it works in the `get_table_rows` method.

Reply:
Yeah, I like your wild idea. To avoid confusion, maybe we just use `maxRows` in the `query()` and `first()` methods and we allow developers to specify `perRequestLimit` in the `next()` call.

Reply:
If we want to go on with removing `limit`, let's try to figure out how name those both to something more specific (and specifically for advanced use cases) to be close to accurate, and get them both in any `Table` call that returns a `Cursor`. 

Then the cursor itself will use that as the "Maximum rows per API Request" value. 

Then yeah, the `next()` call on a cursor could override that value for the fetch that's about to happen.

For naming... maybe since it's probably going to be a very rarely used parameter, let's get super specific. Maybe it's something like `rowsPerAPIRequest` or something? 😂 

```ts
const cursor = table.query({
    from: 5, 
    to: 10, 
    rowsPerAPIRequest: 10 // defaults to 1000, this overrides to 10
})

await cursor.next() // gives me 10 rows
await cursor.next() // gives me the next 10 rows
```

The developer could also achieve the same results with:

```ts
const cursor = table.query({
    from: 5, 
    to: 10, 
})

await cursor.next(10) // gives me 10 rows
await cursor.next(10) // gives me the next 10 rows
```

Reply:
I guess we could also let them specify `perRequestLimit` in the `query()`and `first()` methods 🤔

Reply:
I'm not sure I follow your second example (isn't the default 1000 rows??), but apart from that it all makes sense! I can add those changes to this PR tomorrow.

Reply:
Ah yes, I'm sorry... that was a bad copy paste job 😂 

The 2nd example should be:

```ts
const cursor = table.query({
    from: 5, 
    to: 10, 
})

await cursor.next(10) // gives me up to 10 rows, 1 API request
await cursor.next(10) // gives me up to the next 10 rows, 1 API request
```

It was just showing you could pass `10` into each `next()` call to achieve the same results as if you defined `rowsPerAPIRequest: 10`. Up to the developer to choose which they want.

Reply:
haha yeah, makes sense now 😄